hudi-spark-datasource/hudi-spark3/pom.xml

```diff
@@ -157,7 +175,7 @@
     <dependency>
       <groupId>org.apache.spark</groupId>
       <artifactId>spark-sql_2.12</artifactId>
-      <version>${spark3.version}</version>
+      <version>${spark3.2.version}</version>
```

Comment:
can you clarify why this change? what if user needs to build the project with spark 3.1.x profile?

Reply:
> can you clarify why this change? what if user needs to build the project with spark 3.1.x profile?

spark 3.1.x  profile use `<hudi.spark.module>hudi-spark3.1.x</hudi.spark.module>` with spark3.1.x version.
spark 3  profile use `<hudi.spark.module>hudi-spark3</hudi.spark.module>` with spark 3.2.0(and above) versions.

Reply:
we should prefer to simple set of properties to maintain: both `spark3.version` and `hudi.spark.module` change based on different spark profile. When `spark3.version` switch to `3.1.2`, module will be `hudi-spark3.1.x` and effectively ignores `hudi-spark3` here. So I don't think we need to introduce more properties here.